### PR TITLE
PAY-1128 Fetch consent using actor details and build filtering rules for consent-based filtering

### PR DESCRIPTION
**Pull request by nirbhaykumar1999**
## Summary
- Add `DelegatedActorRulesManager` to fetch active Consent for grantor/grantee pair and parse denied sensitive categories from nested provisions
- Add `FilteringRulesCacheKeyGenerator` for generation-based Redis caching of filtering rules
- Add `DelegatedActorScopeManager` to gate access based on valid consent
- Refactor `ScopesValidator` to async with delegated actor consent check
- Add consent policy URI to delegated actor audit event agent

## Test plan
- [x] 17 unit tests for DelegatedActorRulesManager (consent fetching, filtering rule parsing, caching, edge cases)
- [x] Verify no regressions in existing JWT strategy and audit log tests

## Testing Step (Local)
- Get Reference to test cases and created different types of consent and only one should be active at a time
- Validate error is thrown is no consent or more consent are found. 
- Using Debug Mode, Validate the filtering rules are build correcrtly

### CODE REVIEW

File: src/utils/configManager.js [hunk 1]
```diff
@@ -1133,6 +1133,14 @@ class ConfigManager {
     get validateDelegatedAccessToken() {
         return isTrue(env.VALIDATE_DELEGATED_ACCESS_TOKEN);
     }
+
+    get sensitiveCategorySystemIdentifier() {
```
[the remark this answers is not in the document]
Comment (nirbhaykumar1999) [reply src/utils/configManager.js]: it will be a major refactoring change, we can plan this later..not a part of this PR

File: src/operations/security/delegatedActorScopeManager.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+     */
+    async isAccessAllowedAsync({delegatedActor, personIdFromJwtToken, base_version}) {
+        if (!this.delegatedActorRulesManager.isUserDelegatedActor({delegatedActor})) {
+            return true;
+        }
+
+        return await this.delegatedActorRulesManager.hasValidConsentAsync({
+            delegatedActor,
+            personIdFromJwtToken,
+            base_version
```
[the remark this answers is not in the document]
Comment (nirbhaykumar1999) [reply src/operations/security/delegatedActorScopeManager.js]: These should be handled when building the JWT using consent
